### PR TITLE
add trailers to active_stream record to send when body is done

### PR DESCRIPTION
**Pull request by tsloughter**
Least sure about this one but opening. I haven't looked at it in a while, used it for grpc, so will give it a review myself tomorrow.

### CODE REVIEW

File: src/h2_connection.erl [hunk 1]
```diff
@@ -1000,7 +1012,58 @@ handle_event(_, {send_headers, StreamId, Headers, Opts},
...
+                                           (Conn#connection.peer_settings)#settings.max_frame_size,
+                                           true
+                                          ),
+            OldBody = h2_stream_set:queued_data(Stream),
+
+            NewS = h2_stream_set:update_trailers(FramesToSend, Stream),
+            NewBody = case is_binary(OldBody) of
+                          true -> OldBody;
+                          false -> <<>>
+                      end,
```
Comment (talentdeficit) [inline src/h2_connection.erl]: in what situations would `OldBody` not be a binary and why replace it in that case?

Comment (tsloughter) [reply src/h2_connection.erl]: `queued_data` can be `undefined` or `done`. So if it is one of those we just make an empty body to send the trailers.

Comment (tsloughter) [reply src/h2_connection.erl]: Hm, this may not be needed.

**Comment by tsloughter** [comment]
@talentdeficit yea, that wasn't needed, I've updated the code.

**Comment by talentdeficit** [comment]
there's one commented out line that should probably be stripped but other than that this looks good

**Comment by tsloughter** [comment]
Oh right, I actually take that out in the lager PR, so won't bother dealing with it in this.

**Pull request merged**